app/models/order_item.rb

```diff
@@ -36,12 +50,13 @@ def item_price # v. price / cost / item_cost / total_item_price / total_item_cos
 
   def product_has_stock? # !Q is this the right way to do this?
     # stock = product.has_available_stock?
-    unless product.has_available_stock?
+    if product.has_available_stock?
```

Comment:
I like the #optimize tag you added. That seems like the best approach here. I was envisioning some sort of validation or something, rather than a before_save or whatever. But this is fine for now.